flocker/node/agents/gce.py

```diff
@@ -891,11 +899,27 @@ def _do_blocking_operation(self,
         :returns dict: A dict representing the concluded GCE operation
             resource.
         """
+        if sleep is None:
+            sleep = time.sleep
+
+        # A custom sleep function that drops the lock while the actual sleeping
+        # is going on.
+        def lock_dropped_sleep(*args, **kwargs):
+            self._lock.release()
+            try:
+                return sleep(*args, **kwargs)
+            finally:
+                self._lock.acquire()
+
         args = dict(project=self._project, zone=self._zone)
         args.update(kwargs)
-        operation = function(**args).execute()
-        return wait_for_operation(
-            self._compute, operation, [1]*timeout_sec, sleep)
+        self._lock.acquire()
+        try:
+            operation = function(**args).execute()
+            return wait_for_operation(
+                self._compute, operation, [1]*timeout_sec, lock_dropped_sleep)
+        finally:
+            self._lock.release()
```

Comment:
:grey_question: The code would be a littler shorter (clearer?) if you used a `with` statement instead of `try...finally...`
 * https://docs.python.org/2/library/threading.html#using-locks-conditions-and-semaphores-in-the-with-statement